### ISSUE TITLE
Can't close file without overwriting

### ISSUE THREAD

**Issue by AliRain** [microsoft/vscode#125405]
Issue Type: <b>Bug</b>

1. Open a file with some git history
2. Make changes to the file without saving
3. Change to another git branch
4. Close this file
5. It'll ask if you want to save, click discard changes
6. The file appears again in the same state as if you didn't close the file.

The only way to get out of this state is to save changes, overwrite, then go to git and undo changes.

VS Code version: Code 1.56.2 (054a9295330880ed74ceaedda236253b4f39a335, 2021-05-12T17:44:30.902Z)
OS version: Darwin x64 20.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|21, 13, 7|
|Memory (System)|16.00GB (0.47GB free)|
|Process Argv|--crash-reporter-id 9f6dd278-1555-46d8-a047-d7043ede85db|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (23)</summary>

Extension|Author (truncated)|Version
---|---|---
htmltagwrap|bra|0.0.7
vscode-postgres|cko|1.1.17
bracket-pair-colorizer-2|Coe|0.2.0
vscode-eslint|dba|2.1.20
githistory|don|0.6.16
es7-react-js-snippets|dsz|3.1.1
gitlens|eam|11.4.1
carbon-now-sh|eri|1.2.0
json-tools|eri|1.0.2
prettier-vscode|esb|6.4.0
vscode-open-multiple-files|fab|1.4.0
auto-close-tag|for|0.5.10
auto-rename-tag|for|0.1.6
vscode-env|Iro|0.1.0
jbockle-format-files|jbo|3.3.0
vscode-styled-components|jpo|1.6.4
vsliveshare|ms-|1.0.4360
color-highlight|nau|2.3.0
vscode-jest|Ort|4.0.2
vscode-paste-and-indent|Rub|0.0.8
sass-indented|syl|1.8.16
vscode-icons|vsc|11.4.0
vscode-import-cost|wix|2.15.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
vspre833cf:30267465
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
pythondataviewer:30285071
vscus158cf:30286554
vscgsv2:30307504
vscorehov:30309549
bridgeflightcf:30302070
vscod805:30301674

```

</details>

**Comment by bpasero** [comment]
> Change to another git branch

How?

```
~/Desktop/test-ts> git co changes
error: Your local changes to the following files would be overwritten by checkout:
        CHANGELOG.md
Please commit your changes or stash them before you switch branches.
Aborting
~/Desktop/test-ts>
```

**Comment by AliRain** [comment]
@bpasero here's a loom video showing the issue: https://www.loom.com/share/627abd4d8eaf49179fb8ea4742ac4d93

The reason it takes so long switching branches is that we have a Husky hook to download yarn dependencies when checking out another branch.

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by AliRain** [comment]
@bpasero I downloaded the latest nightly, installed my most essential extensions: eslint, prettier, styled components. I tested the same workflow, same branches, same file being modified, etc. trying as best as I can to replicate exactly the same steps. I did not face the issue in the nightly build. So I'm guessing that puts in one of the following 3 scenarios:
1. The bug has been fixed in the latest nightly
2. The bug was caused by an extension that I didn't install on the nightly build
3. The bug was caused by some of my custom settings in VS Code

Here's to hoping that it's the first scenario! 😅
